tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -129,12 +137,33 @@ def tags_route(self, request):
     @wrappers.Request.application
     def scalars_route(self, request):
         """Given a tag and single run, return array of ScalarEvents."""
-        tag = request.args.get("tag")
-        run = request.args.get("run")
+        if request.method == "GET":
+            tag = request.args.get("tag")
+            run = request.args.get("run")
+            runs = [run]
+        else:
+            tag = request.form["tag"]
+            json_runs = request.form["runs"]
+            try:
+                runs = json.loads(json_runs)
+            except Exception as e:  # pylint: disable=broad-except
+                # Different JSON libs raise different exceptions, so we just do a
+                # catch-all here. This problem is complicated by how Tensorboard might be
+                # run in many different environments, as it is open-source.
+                # TODO(@caisq, @chihuahua): Create platform-dependent adapter to catch
+                # specific types of exceptions, instead of the broad catching here.
```

Comment:
I see that this is cribbed from the debugger plugin; do we know whether
this is still an issue? Historically, Python 2 and 3 raised different
errors here (`ValueError` vs. the new `JSONDecodeError`), but if that
was the only issue then we should be fine to do the sane thing now. Is
there some terrifying Google-internal `json` monkey patching going on?
(If so, a Buganizer link would be nice.)

As written, this `Exception` also catches, e.g., `NameError`s and other
things that should clearly be 500s, as well as control flow like
`KeyboardInterrupt`s, so I’d like to avoid it if possible.
<https://google.github.io/styleguide/pyguide.html#244-decision>